src/Yaapii.Atoms/Scalar/BitAt.cs

```diff
@@ -0,0 +1,222 @@
+            bytes,
+            position,
+            (exception, bts) => fallback.Invoke(bts)
+        )
+        { }
+
+        /// <summary>
+        /// The value of a particular bit.
+        /// </summary>
+        /// <param name="bytes">Bytes from where the bit is taken</param>
+        /// <param name="position">Zero based bit index in the bytes</param>
+        /// <param name="fallback">Result in case of an error</param>
+        public BitAt(IBytes bytes, int position, Func<IBytes, bool> fallback) : this(
+            bytes,
+            position,
+            new BiFuncOf<Exception, IBytes, bool>((exception, bts) => fallback.Invoke(bts))
+        )
+        { }
+
+        /// <summary>
+        /// The value of a particular bit.
+        /// </summary>
+        /// <param name="bytes">Bytes from where the bit is taken</param>
+        /// <param name="position">Zero based bit index in the bytes</param>
+        /// <param name="fallback">Result in case of an error</param>
+        public BitAt(IBytes bytes, int position, Func<Exception, IBytes, bool> fallback) : this(
+            bytes,
+            position,
+            new BiFuncOf<Exception, IBytes, bool>((exception, bts) => fallback.Invoke(exception, bts))
+        )
+        { }
+
+        /// <summary>
+        /// The value of a particular bit.
+        /// </summary>
+        /// <param name="bytes">Bytes from where the bit is taken</param>
+        /// <param name="position">Zero based bit index in the bytes</param>
+        /// <param name="fallback">Result in case of an error</param>
+        public BitAt(IBytes bytes, int position, IBiFunc<Exception, IBytes, bool> fallback)
+        {
+            this.bytes = bytes;
+            this.position = position;
+            this.fallback = fallback;
+        }
+
+        public bool Value()
+        {
+            bool result;
+
+            FailWhenIndexIsNegative();
+            var byteIndex = this.position / 8;
+            var bitInByteIndex = this.position % 8;
+            if (this.CanGetBit(byteIndex))
+            {
+                var relevantByte = this.bytes.AsBytes()[byteIndex];
+                result = (relevantByte & (1 << bitInByteIndex)) > 0;
+            }
+            else
+            {
+                result =
+                    this.fallback.Invoke(
+                        new ArgumentException($"Cannot get byte at position {byteIndex} because there are only {this.bytes.AsBytes().Length} bytes."),
+                        this.bytes
+                    );
+            }
+
+            return result;
+        }
+
+        private void FailWhenIndexIsNegative()
+        {
+            new FailWhen(() =>
+                this.position < 0,
+                new ArgumentException($"The position must be non-negative but is {this.position}")
+            ).Go();
+        }
+
+        private bool CanGetBit(int byteIndex)
+        {
+            return this.bytes.AsBytes().Length > byteIndex;
```

Comment:
You are unnecessarily building the bytes two times inside this object. Please refactor.